### DISCUSSION TITLE
Lesson 6 - Metamask unable to fetch Chain ID to enable creation of localhost

### DISCUSSION THREAD

**Discussion by yeahChibyke** [Cyfrin/foundry-full-course-cu#425] [Q&A] [answered]
Metamask keeps telling me 'Could not fetch fetch Chain ID...'. I inputted the RPC link correctly from anvil. And by the way, I am using Gitpod.
![local_blockchain_error_1](https://github.com/Cyfrin/foundry-full-course-f23/assets/56419077/81225005-7b34-48eb-a7b9-01808d5dc386)

**Comment by ghost** [comment]
Hello @yeahChibyke, run `anvil` and try again

**Reply by yeahChibyke** [reply]
Currently doing that. Still getting the same issue

**Reply by ghost** [reply]
I've never used Gitpod but that's the RPC URL you're getting from anvil?

**Reply by yeahChibyke** [reply]
Yes. The exact same RPC URL

**Comment by 0xkumar** [comment]
Make sure that ANVIL chain is running while you creating it.

**Reply by yeahChibyke** [reply]
Currently doing that, still getting the same issue. I currently have anvil running in gitpod, yet I keep getting a 'Chain ID' error

**Reply by olahamid** [reply]
hello @yeahChibyke . i got the same issue, just worked out fine for me. try connecting gitpod to VScode, run anvil in your terminal then try adding on metamask.

**Comment by Jab-11** [comment, marked as answer]
Hi, @yeahChibyke. I had the same problem. What you can do is run Anvil in a separate terminal. You will find a port section next to the terminal section. Open that section and copy the URL from there, which is the RPC URL.

**Reply by theirrationalone** [reply]
Delete registered network if you already have one with chainid 31337....

Or Just close metamask and try again, again again...

**Reply by yeahChibyke** [reply]
I copied a URL which looks something like this:
https://8545-yeahchibyke-myfoundrysi-5u1fd5s2tyv.ws-eu102.gitpod.io

Do I still use the same Chain ID - 31337? Because I am still getting the same 'invalid RPC URL' error.

**Reply by yeahChibyke** [reply]
It worked. I made the port public. Thanks a lot. You are so wonderful
